home/tests/test_serializers.py

```diff
@@ -0,0 +1,258 @@
+        wa_body = f"*Default WhatsApp Content {i+1}* 🏥"
+
+        variation_messages = [
+            VarMsg(
+                wa_body.replace("Default", f"Varied {var+1}"),
+                gender=gender_variations[var],
+            )
+            for var in range(wa_var)
+        ]
+
+        bodies.append(
+            WABody(
+                title,
+                [
+                    WABlk(
+                        wa_body,
+                        variation_messages=variation_messages,
+                    )
+                ],
+            )
+        )
+
+    content_page = PageBuilder.build_cp(
+        parent=parent,
+        slug=title.replace(" ", "-"),
+        title=title,
+        bodies=bodies,
+        tags=[],
+        quick_replies=[],
+        triggers=[],
+        publish=True,
+    )
+    return content_page
+
+
+@pytest.mark.django_db
+class TestHasNextMessage:
+    # Test has_next_message
+    def test_next_message_whatsapp(self) -> None:
+        """Checks if the next message matches up.
+        Index of first message is 0, next message is then message 2 with index 1"""
+        page = create_content_page(wa_body_count=2, wa_var=0)
+        assert page.whatsapp_body
+        assert has_next_message(0, page, "whatsapp") == 2
+
+    def test_no_next_message_for_viber(self) -> None:
+        """On a page with 2 viber messages, the next message is None"""
+        page = create_content_page()
+        assert page.viber_body
+        assert has_next_message(2, page, "viber") is None
+
+    def test_next_message_for_viber(self) -> None:
+        """On a page with 2 viber messages, the next message is None"""
+        page = create_content_page()
+        assert page.viber_body
+        assert has_next_message(1, page, "viber") is None
+
+    def test_no_next_message_for_invalid_platform(self) -> None:
+        """Returns none on a platform that is unrecognised"""
+        page = create_content_page()
+        assert has_next_message(0, page, "email") is None
+
+    def test_no_next_message_for_null_content_page(self) -> None:
+        """has_next_message cannot be called on None"""
+        with pytest.raises(AttributeError):
+            has_next_message(0, None, "whatsapp")
+
+    def test_no_next_message_empty_body(self) -> None:
+        """No next message on a page with no whatsapp body"""
+        page = create_content_page(wa_body_count=0)
+        assert has_next_message(0, page, "whatsapp") is None
+
+    def test_no_next_message_on_last_message(self) -> None:
+        """last messages' next message is None"""
+        page = create_content_page(wa_body_count=1)
+        assert has_next_message(1, page, "whatsapp") is None
+
+    def test_next_message_on_first_message_of_many(self) -> None:
+        """Page with many whatsapp messages has a next message on the first"""
+        page = create_content_page(wa_body_count=5)
+        assert has_next_message(0, page, "whatsapp") == 2
```

Comment:
Mixing zero-indexed (for the input) and one-indexed (for the output) values creates all kinds of confusion. We should probably fix the code under test to be consistent. (That's a job for another PR, though.)